recipes/cluster-controller.rb

```diff
@@ -30,6 +37,29 @@
   yum_package "dhcp"
 else
   include_recipe "eucalyptus::install-source"
+  eucalyptus_dir = source_directory
+  if node['eucalyptus']['source-repo'].end_with?("internal")
+    eucalyptus_dir = "#{source_directory}/eucalyptus"
+  end
+  if Chef::VersionConstraint.new("~> 6.0").include?(node['platform_version'])
+    tools_dir = "#{eucalyptus_dir}/tools"
+    execute "ln -sf #{tools_dir}/eucalyptus-cloud /etc/init.d/eucalyptus-cc" do
+      creates "/etc/init.d/eucalyptus-cc"
+    end
+    execute "chmod +x #{tools_dir}/eucalyptus-cloud"
+  end
+  if Chef::VersionConstraint.new("~> 7.0").include?(node['platform_version'])
+    file '/usr/lib/systemd/system/eucalyptus-cluster.service' do
+      lazy { content IO.read("#{eucalyptus_dir}/systemd/units/eucalyptus-cluster.service") }
+      mode '0644'
+      action :create
+      not_if do ::File.exists?('/usr/lib/systemd/system/eucalyptus-cluster.service') end
+    end
+    execute "ln -sf /usr/lib/systemd/system/eucalyptus-cluster.service /usr/lib/systemd/system/eucalyptus-cc.service" do
+      creates "/usr/lib/systemd/system/eucalyptus-cc.service"
+      not_if do ::File.exists?('/usr/lib/systemd/system/eucalyptus-cc.service') end
+    end
+  end
```

Comment:
I'm a little confused as to why this would need to do this at all.  If it supplies the correct options to configure then make install is supposed to install all these files itself.